packages/medusa/src/api/store/products/middlewares.ts

```diff
@@ -57,6 +58,7 @@ export const storeProductRoutesMiddlewares: MiddlewareRoute[] = [
         StoreGetProductsParams,
         QueryConfig.retrieveProductQueryConfig
       ),
+      applyParamsAsFilters({ id: "id" }),
```

Comment:
There's already a `applyDefaultFilters` that you can use for this.

Reply:
Ah I guess I can use the function there, let me switch 👍

Reply:
The thing is, the filter value comes from the path params, and currently you can't do that with `applyDefaultFilters`. I think having this makes sense as it has different semantics